### ISSUE TITLE
post-type taxonomy appears in get_taxonomies() output without proper labels

### ISSUE THREAD

**Issue by benlk** [INN/Largo#1173]
This happens only if the option to enable the post-types prominence term is unchecked in Theme Options > Advanced > Enable "Post Types" taxonomy.

If the option is unchecked, the label of the post-type taxonomy as rendered in the Largo Taxonomy List widget is "Tag":
<img width="400" alt="screen shot 2016-03-17 at 11 59 21 am" src="https://cloud.githubusercontent.com/assets/1754187/13852286/cdee1302-ec37-11e5-90ed-938504e95f03.png">

If the option is checked, the label is the correct "Post Types":
<img width="400" alt="screen shot 2016-03-17 at 11 58 53 am" src="https://cloud.githubusercontent.com/assets/1754187/13852304/dfdb76f4-ec37-11e5-9174-bb799a248c03.png">

I think this is because the action that hides the taxonomy is stripping all the labels: https://github.com/INN/Largo/blob/74508518af57e99a378cdf8fd31317c56667118f/inc/taxonomies.php#L466-L471

```php
	if ( of_get_option('post_types_enabled') == 0 ) {
		register_taxonomy( 'post-type', array(), array('show_in_nav_menus' => false) );
	}
```

So the solution is to either change [the initial registration function in inc/term-icons.php](https://github.com/INN/Largo/blob/f4ddf0b27b5fc34fb826e7ac22a9c393b4cbed1a/inc/term-icons.php#L32) to set `show_in_nav_menus` on registration, or to change the reregistration function to not obliterate the labels.

**Comment by benlk** [comment]
IMHO, this should be tackled before 0.5.5 ships with #1171. It will cause confusion otherwise: Users will select the post-types "Tag" entry and wonder why no tags appear.

**Comment by benlk** [comment]
The `public` option of the term should also be set to `false` if the taxonomy is meant to be hidden.

**Comment by aschweigert** [comment]
People are literally only going to use this widget for series (and maybe categories) as I've said before :)

But, I do agree with the general point that we should set these options correctly when all custom taxonomies are registered. Also, the post types taxonomy should be registered in inc/taxonomies with everything else and not in inc/term-icons

**Comment by benlk** [comment]
Moving to hotfix for [HELPDESK-589](http://jira.inn.org/browse/HELPDESK-589), will include this fix in #1171.

**Comment by benlk** [comment]
#1171 is closed; this fix is now in #1187 targeting 0.5.5.
